pkg/api/v1beta1/conversion.go

```diff
@@ -1279,32 +1300,10 @@ func init() {
 			if err := s.Convert(&in.TypeMeta, &out.ObjectMeta, 0); err != nil {
 				return err
 			}
-			if err := s.Convert(&in.Protocol, &out.Protocol, 0); err != nil {
+			if err := s.Convert(&in.Subsets, &out.Subsets, 0); err != nil {
 				return err
 			}
-			for i := range in.Endpoints {
-				out.Endpoints = append(out.Endpoints, newer.Endpoint{})
-				ep := &out.Endpoints[i]
-				host, port, err := net.SplitHostPort(in.Endpoints[i])
-				if err != nil {
-					return err
-				}
-				ep.IP = host
-				pn, err := strconv.Atoi(port)
-				if err != nil {
-					return err
-				}
-				ep.Port = pn
-				for j := range in.TargetRefs {
-					if in.TargetRefs[j].Endpoint != in.Endpoints[i] {
-						continue
-					}
-					ep.TargetRef = &newer.ObjectReference{}
-					if err := s.Convert(&in.TargetRefs[j].ObjectReference, ep.TargetRef, 0); err != nil {
-						return err
-					}
-				}
-			}
+			// Back-compat fields are handled in the defaulting phase.
```

Comment:
Defaulting runs any time we convert a versioned object into internal, which
should include updates.

On Thu, Mar 26, 2015 at 10:10 PM, Brian Grant <notifications@github.com>
wrote:

> In pkg/api/v1beta1/conversion.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5939#discussion_r27276318>
> :
>
> > -				pn, err := strconv.Atoi(port)
> > -				if err != nil {
> > -					return err
> > -				}
> > -				ep.Port = pn
> > -				for j := range in.TargetRefs {
> > -					if in.TargetRefs[j].Endpoint != in.Endpoints[i] {
> > -						continue
> > -					}
> > -					ep.TargetRef = &newer.ObjectReference{}
> > -					if err := s.Convert(&in.TargetRefs[j].ObjectReference, ep.TargetRef, 0); err != nil {
> > -						return err
> > -					}
> > -				}
> > -			}
> > +			// Back-compat fields are handled in the defaulting phase.
>
> Does the defaulting pass get invoked on the update path, or do you verify
> equality of the two representations?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5939/files#r27276318>
> .
>